test/server/lib/ACLFormula.ts

```diff
@@ -130,14 +167,25 @@ describe('ACLFormula', function() {
     assert.equal(compiled({user: new User({}), rec: V({A: null, B: 2})}), true);
     assert.equal(compiled({user: new User({}), rec: V({A: null, B: false})}), false);
     assert.equal(compiled({user: new User({}), rec: V({A: null, B: 0})}), true);
+  });
 
-    // Test nested attribute lookups.
-    compiled = await setAndCompile('user.office.city == "New York"');
+  it('should handle nested attribute lookups', async function () {
+    const compiled = await setAndCompile('user.office.city == "New York"');
     assert.equal(compiled({user: new User({office: V({city: "New York"})})}), true);
     assert.equal(compiled({user: new User({office: V({city: "Boston"})})}), false);
     assert.equal(compiled({user: new User({office: V({city: null})})}), false);
     assert.throws(() => compiled({user: new User({})}), /No value for 'user.office'/);
     assert.equal(compiled({user: new User({office: 5})}), false);
     assert.throws(() => compiled({user: new User({office: null})}), /No value for 'user.office'/);
   });
-});
+
+  it('should handle "in" and "not in" when RHS is nullish', async function() {
+    let compiled = await setAndCompile('user.Email in rec.emails');
+    const user = new User({Email: 'X@'});
+    assert.equal(compiled({user, rec: V({emails: null})}), false);
```

Comment:
Can you add a `true` test here, just to demonstrate clearly there isn't an accidental test typo causing the `false`s.